src/main/java/org/damap/base/rest/invenio_damap/InvenioDAMAPService.java

```diff
@@ -0,0 +1,120 @@
+package org.damap.base.rest.invenio_damap;
+
+import io.quarkus.security.UnauthorizedException;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.json.JsonObject;
+import jakarta.transaction.Transactional;
+import java.text.MessageFormat;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.damap.base.rest.dmp.domain.DatasetDO;
+import org.damap.base.rest.dmp.domain.DmpDO;
+import org.damap.base.rest.dmp.service.DmpService;
+import org.damap.base.rest.madmp.dto.Dataset;
+import org.damap.base.rest.version.VersionDO;
+import org.damap.base.rest.version.VersionService;
+import org.eclipse.microprofile.jwt.JsonWebToken;
+
+@ApplicationScoped
+public class InvenioDAMAPService {
+
+  protected DmpService dmpService;
+  protected VersionService versionService;
+
+  @Inject
+  InvenioDAMAPService(DmpService dmpService, VersionService versionService) {
+    this.dmpService = dmpService;
+    this.versionService = versionService;
+  }
+
+  // Resolves user identity based on the DMPs and returns dmp list and list of
+  // identifiers if successful.
+  public SimpleEntry<List<DmpDO>, List<String>> resolveDmpsAndIds(JsonWebToken jwt) {
+    JsonObject invenioDamapClaim = jwt.getClaim("invenio-damap");
+    if (invenioDamapClaim == null) {
+      throw new UnauthorizedException("Missing invenio-damap claim in jwt.");
+    }
+
+    JsonObject identifiers = invenioDamapClaim.getJsonObject("identifiers");
+    if (identifiers == null || identifiers.isEmpty()) {
+      throw new UnauthorizedException("No valid authentication schema was provided.");
+    }
+
+    List<DmpDO> personDmpList = null;
+    List<String> matchingIdentifiers = new ArrayList<>();
+
+    for (String key : identifiers.keySet()) {
+      String identifier = identifiers.getString(key);
+      List<DmpDO> dmps = dmpService.getDmpDOListByPersonId(identifier);
+
+      // Check if returned DMPs is null or the list is empty.
+      // If yes, no resolving can take place, move to the next identifier.
+      if (dmps == null || dmps.isEmpty()) continue;
+
+      // Here dmps would contain at least one DMP so keep track of the current
+      // iterated identifier.
+      matchingIdentifiers.add(identifier);
```

Comment:
> From DAMAP side at this level, if we get results for different identifiers, can we assume different identities?
I would say yes, since DAMAP will always use only one type of identifier and invenio will send only different types. So the only way this error could even be triggered, is by some admin making a mistake. So i would propose to throw the unauthorized exception when two identifiers match as a failsafe) and to remove the list equality check, like you suggested.